framework/src/main/java/org/checkerframework/framework/type/AbstractViewpointAdapter.java

```diff
@@ -294,27 +299,31 @@ protected AnnotatedTypeMirror combineAnnotationWithType(
             return declared;
         } else if (declared.getKind() == TypeKind.DECLARED) {
             AnnotatedDeclaredType adt = (AnnotatedDeclaredType) declared.shallowCopy();
```

Comment:
Can this be re-ordered to avoid the creation of the shallow copy if it isn't used at all?